src/mon/OSDMonitor.cc

```diff
@@ -1663,6 +1665,15 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
     for (map<int,failure_reporter_t>::iterator p = fi.reporters.begin();
 	 p != fi.reporters.end();
 	 ++p) {
+      //get subtree reporter
+      map<string, string> reporter_loc = osdmap.crush->get_full_location(p->first);
+      map<string, string>::iterator iter = reporter_loc.find(reporter_subtree_level); 
```

Comment:
Oh, yes, it doesn't.
Would we prefer to change the get_full_location? or just have something like
if (reporter_subtree_level = osd) {
   reporters_by_subtree.insert(p->first)
} else {
    map<string, string> reporter_loc = osdmap.crush->get_full_location(p->first);
    ........
}